.. change::
    :tags: feature, engine
    :tickets: 8710

    To better support the use case of iterating :class:`.Result` and
    :class:`.AsyncResult` objects where user-defined exceptions may interrupt
    the iteration, both objects as well as variants such as
    :class:`.ScalarResult`, :class:`.MappingResult`,
    :class:`.AsyncScalarResult`, :class:`.AsyncMappingResult` now support
    context manager usage, where the result will be closed at the end of
    the context manager block.

    In addition, ensured that all the above
    mentioned :class:`.Result` objects include a :meth:`.Result.close` method
    as well as :attr:`.Result.closed` accessors, including
    :class:`.ScalarResult` and :class:`.MappingResult` which previously did
    not have a ``.close()`` method.

    .. seealso::

        :ref:`change_8710`


.. change::
    :tags: bug, typing

    Corrected various typing issues within the engine and async engine
    packages.
